trestle/core/catalog_interface.py

```diff
@@ -332,11 +333,51 @@ def write_catalog_as_markdown(
     @staticmethod
     def _get_group_ids(md_path: pathlib.Path) -> List[str]:
         """Get the list of group ids from the directories in the markdown path."""
-        group_ids: List[str] = []
+        # need to start with empty list to find controls not in group
+        group_ids: List[str] = ['']
 
         for gdir in md_path.glob('*/'):
-            group_ids.append(str(gdir.stem))
-        return group_ids
+            if gdir.is_dir():
+                group_ids.append(str(gdir.stem))
+        return sorted(group_ids)
+
+    @staticmethod
+    def _get_control_paths(group_path: pathlib.Path) -> List[pathlib.Path]:
```

Comment:
Might be worth trying to leverage: from distutils.version import StrictVersion


or : https://pypi.org/project/natsort/  which looks kidn of cool

Reply:
As discussed, the true order that actually matters is the one in the resolved catalog, so if we really want to enforce order we will need to provide the profile and go by the order in the resolved catalog.  What I've done here is completely safe because if a control name does not have the expected form it will just be sorted alphabetically with the others in a consistent manner.   I will add a comment indicating that.

Reply:
As discussed, the true order that actually matters is the one in the resolved catalog, so if we really want to enforce order we will need to provide the profile and go by the order in the resolved catalog. What I've done here is completely safe because if a control name does not have the expected form it will just be sorted alphabetically with the others in a consistent manner. I will add a comment indicating that.